src/Pelagos/Util/DOIutil.php

```diff
@@ -375,4 +333,56 @@ function ($matches) {
             $input
         );
     }
+
+    /**
+     * This function generates a new DOI string that is not already in use per datacite.
+     *
+     * @return string A unique, unused DOI string.
+     */
+    public function generateDoi()
+    {
+        $encoder = new Base32CrockfordEncoder();
+        // 1099511627775 encodes to the longest 8 character Crockford 32 string.
+        $max = 1099511627775;
+        // Start at 1 (0 is problematic as library does not produce checksum for 0.)
+        $random = mt_rand(1, $max);
+        // Add shoulder and remove the checksum character.
+        $doi = $this->doishoulder . '/' . substr($encoder->encode($random), 0, -1);
+        while ($this->checkDoiExists($doi)) {
+            $random = mt_rand(0, $max);
+            $doi = $this->doishoulder . '/' . substr($encoder->encode($random), 0, -1);
+        }
+        return $doi;
+    }
+
+    /**
+     * This function queries Datacite to determine if the DOI is already in use.
+     *
+     * @param string $doi The DOI to be checked.
+
+     * @throws HttpClientErrorException When there is a 4xx error negotiating with Datacite, other than a 400.
+     * @throws HttpServerErrorException When there is a 5xx error negotiating with Datacite.
+     *
+     * @return boolean True if returned if the DOI exists, false if it does not.
+     */
+    protected function checkDoiExists($doi)
```

Comment:
I feel checking the doi on datacite is overkill. If we still want to check the DOI exists on datacite, we can use the getDoiMetadata method to verify it doesn't exist. Similar method exists in DoiConsumer.php (doiAlreadyExists() mehtod)

Reply:
I tried using the getDoiMetadata method, as also Mike suggested, but since we are having to rely specifically on catching a 400 for detectection, I thought we'd end up relying on a catch as our detection method. I'll take a look at the DoiConsumer's method though.

Reply:
I replaced with a call to getDOIMetadata(), however, the check is still needed as the one made by the consumer will consider a collision an existing DOI, which is an error, where is here we just want to be able to pick a unique DOI.